### PR TITLE
Make MockWebServer directly usable as a rule.

### PR DESCRIPTION
**Pull request by JakeWharton**
This eliminates the awkward sometimes-calls to get() in order to access APIs which were not exposed directly on the rule.

### CODE REVIEW

**Comment by JakeWharton** [comment]
In a case of the very odd... Travis CI passed this on JDK 7 but not JDK 8. This is because the 'Sample Server' example uses MockWebServer but does not have a dependency on JUnit.

I guess the advantage to a separate class was that the JUnit dependency was optional.

What do we feel here? Is MockWebServer testing only and have a JUnit dependency or should it be able to stand on its own without it?

**Comment by swankjesse** [comment]
Making MockWebServer depend on JUnit is fine by me.

Here's one option for MockWebServer In 2015:

 * New `okhttp-testing` module that uses interceptor to make setting up scenarios and confirming interactions possible. ([issue 1096](https://github.com/square/okhttp/issues/1096)). Probably with a new API.
 * MockWebServer degraded to be test support for OkHttp's own implementation.
 * Possibly permit MockWebServer to also implement the `okhttp-testing` API as a real server. For example, for unit testing vs. fuctional testing.

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java [hunk 1]
```diff
@@ -122,17 +132,17 @@ public void setServerSocketFactory(ServerSocketFactory serverSocketFactory) {
   }
 
   public int getPort() {
-    if (port == -1) throw new IllegalStateException("Call play() before getPort()");
+    startOrThrow();
```
Comment (swankjesse) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: awww, does it really have to be this automatic? I liked the explicit lifecycle

**Comment by swankjesse** [comment]
LGTM

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java (hunk 1 above)
Comment (JakeWharton) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: The explicit life cycle is maintained for non-rule usage. I think we can
kill this though.

Comment (swankjesse) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: Yeah, let's go 100% in on the rule. The `play`/`shutdown` names aren't necessary. That comes from an earlier time, when you needed to provide all the responses before you called `play`, and it was intended to offer a record/replay experience.

**Comment by JakeWharton** [comment]
Having second thoughts. Low value prop here. This can wait until we have the scriptable mock interceptor.

**Pull request closed without merging**